pkgs/by-name/wa/warehouse/package.nix

```
@@ -0,0 +1,69 @@
+{
+  lib,
+  blueprint-compiler,
+  desktop-file-utils,
+  fetchFromGitHub,
+  flatpak,
+  flatpak-xdg-utils,
+  glib,
+  gobject-introspection,
+  gtk4,
+  libadwaita,
+  meson,
+  ninja,
+  pkg-config,
+  python3,
+  stdenv,
+  wrapGAppsHook4,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "warehouse";
+  version = "2.0.2";
+
+  src = fetchFromGitHub {
+    owner = "flattool";
+    repo = "warehouse";
+    rev = finalAttrs.version;
+    hash = "sha256-EcpHFS0EczUDFs0A/7IuNs1082hsuuS0J6wxSq47vaQ=";
+  };
+
+  nativeBuildInputs = [
+    blueprint-compiler
+    desktop-file-utils
+    meson
+    ninja
+    pkg-config
+    wrapGAppsHook4
+    gobject-introspection
+  ];
+
+  buildInputs = [
+    flatpak
+    flatpak-xdg-utils
+    glib
+    gtk4
+    libadwaita
+    (python3.withPackages (
+      ps: with ps; [
+        pygobject3
+      ]
+    ))
+  ];
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix PATH : "${flatpak-xdg-utils}/bin"
```

```suggestion
      --prefix PATH : ${lib.makeBinPath [ flatpak-xdg-utils ]}
```

You might also want to consider using `--suffix` instead of `--prefix` to make it easier for users to use their own instances of these programs if they already have them in their `PATH`.